search/includes/classes/class-search.php

```diff
@@ -1482,6 +1489,26 @@ public function filter__ep_prepare_meta_allowed_protected_keys( $keys, $post ) {
 		return \apply_filters( 'vip_search_post_meta_allow_list', $keys, $post );
 	}
 
+	/**
+	 * Since we've established that enabling the protected content feature causes attachments
+	 * to be indexed, we should ensure that 'attachment' is in the indexable post types if
+	 * protected content is enabled.
+	 *
+	 * @param array $indexable_post_types Current list indexable post types in VIP Search.
+	 * @return array New list of indexable post types in VIP Search.
+	 */
+	public function add_attachment_to_ep_indexable_post_types( $indexable_post_types ) {
+		if ( ! is_array( $indexable_post_types ) ) {
+			return $indexable_post_types;
+		}
+		
+		if ( ! in_array( 'attachment', $indexable_post_types, true ) ) {
+			$indexable_post_types[] = 'attachment';
```

Comment:
This array is actually an associative array, here's an example: https://github.com/Automattic/ElasticPress/blob/4ac57414105483818c2ad6464e50547248f49a20/includes/classes/Feature/Documents/Documents.php#L81